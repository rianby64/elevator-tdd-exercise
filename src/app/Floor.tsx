
import * as React from 'react';
import { FloorPanel } from './FloorPanel';

interface FloorProps {
    level: number;
    firstLevel?: boolean;
    lastLevel?: boolean;
}

interface DefaultProps {
    firstLevel: boolean;
    lastLevel: boolean;
}

export class Floor extends React.Component<FloorProps> {

    public static defaultProps: DefaultProps = {
        firstLevel: false,
        lastLevel: false,
    }

    public render(): JSX.Element {
        return (
            <div>
                <span>Floor {this.props.level}</span>
                <FloorPanel
                    upButton={!this.props.lastLevel}
                    downButton={!this.props.firstLevel} />
            </div>
        );
    }
}
