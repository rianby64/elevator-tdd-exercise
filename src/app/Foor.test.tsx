
import * as React from 'react';
import { mount } from 'enzyme';
import { Floor } from './Floor';

test('Floor exists', (): void => {
    const el = (<Floor level={1} />);
    const sh = mount(el);
    expect(sh).toMatchSnapshot();
});

test('Floor is at the first level', (): void => {
    const el = (<Floor level={1} firstLevel={true} />);
    const sh = mount(el);
    expect(sh).toMatchSnapshot();
});

test('Floor is at the last level', (): void => {
    const el = (<Floor level={1} lastLevel={true} />);
    const sh = mount(el);
    expect(sh).toMatchSnapshot();
});
