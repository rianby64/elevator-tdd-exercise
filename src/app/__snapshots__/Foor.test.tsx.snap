// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Floor exists 1`] = `
<Floor
  firstLevel={false}
  lastLevel={false}
  level={1}
>
  <div>
    <span>
      Floor 
      1
    </span>
    <Component
      downButton={true}
      upButton={true}
    >
      <ul>
        <li>
          Up
        </li>
        <li>
          Down
        </li>
      </ul>
    </Component>
  </div>
</Floor>
`;

exports[`Floor is at the first level 1`] = `
<Floor
  firstLevel={true}
  lastLevel={false}
  level={1}
>
  <div>
    <span>
      Floor 
      1
    </span>
    <Component
      downButton={false}
      upButton={true}
    >
      <ul>
        <li>
          Up
        </li>
      </ul>
    </Component>
  </div>
</Floor>
`;

exports[`Floor is at the last level 1`] = `
<Floor
  firstLevel={false}
  lastLevel={true}
  level={1}
>
  <div>
    <span>
      Floor 
      1
    </span>
    <Component
      downButton={true}
      upButton={false}
    >
      <ul>
        <li>
          Down
        </li>
      </ul>
    </Component>
  </div>
</Floor>
`;
